test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -349,3 +352,148 @@
+                                       :disable-infra-crl false })
+                   infra-crl-response-body (:body infra-crl-response)
+                   full-crl-response (handle-get-certificate-revocation-list
+                              request {:cacrl (str master-conf-dir "/ssl/ca/ca_crl.pem") 
+                                       :infra-crl-path (str master-conf-dir "/ssl/ca/infra-crl.pem")
+                                       :disable-infra-crl true })
+                   full-crl-response-body (:body full-crl-response)]
+               (is (map? infra-crl-response))
+               (is (= 200 (:status infra-crl-response)))
+               (is (= "text/plain" (get-in infra-crl-response [:headers "Content-Type"])))
+               (is (and (string? infra-crl-response-body) (= infra-crl-response-body (slurp (str master-conf-dir "/ssl/ca/infra-crl.pem")))))
+               (is (map? full-crl-response))
+               (is (= 200 (:status full-crl-response)))
+               (is (= "text/plain" (get-in full-crl-response [:headers "Content-Type"])))
+               (is (and (string? full-crl-response-body) (= full-crl-response-body (slurp (str master-conf-dir "/ssl/ca/ca_crl.pem")))))
+          )))
+
+        (bootstrap/with-puppetserver-running-with-mock-jrubies
+         "JRuby mocking is safe here because all of the requests are to the CA
+         endpots, which are implemented in Clojure."
+          app
+          {:jruby-puppet {:master-conf-dir master-conf-dir}
+           :certificate-authority {:compile-masters ["compile-master"]
+                                   :disable-infra-crl false}}
+          (testing "Verify infrastructure CRL is returned "
+            (let [options {:ssl-ca-cert (str master-conf-dir "/ssl/ca/ca_crt.pem")
+                           :as :text}
+                  crl-response (http-client/get
+                                   "https://localhost:8140/puppet-ca/v1/certificate_revocation_list/ca"
+                                   (merge options {:headers {"Accept" "text/plain"}}))
+                  crl-response-body (:body crl-response)]
+              (is (map? crl-response))
+              (is (= 200 (:status crl-response)))
+              ;; (is (= "text/plain" (get-in crl-response [:headers "Content-Type"])))  ;; CA does not return this
+              (is (and (string? crl-response-body) (= crl-response-body (slurp (str master-conf-dir "/ssl/ca/infra-crl.pem")))))
+          )))
+
+        (bootstrap/with-puppetserver-running-with-mock-jrubies
+         "JRuby mocking is safe here because all of the requests are to the CA
+         endpots, which are implemented in Clojure."
+          app
+          {:jruby-puppet {:master-conf-dir master-conf-dir}
+           :certificate-authority {:compile-masters ["compile-master"]
+                                   :disable-infra-crl true}}
+          (testing "Verify full CRL is returned "
```

Comment:
How are these last two tests different from the `"Verify correct CRL is returned depending on disable-infra-crl"` test above?

Reply:
The last two verify correct crl is returned over http endpoints (more of an end to end tests). The ones above them directly invoke handle-get-certificate-revocation-list. This is per the multiple levels of tests you had suggested.